samples/api/oit_linked_lists/oit_linked_lists.cpp

```diff
@@ -0,0 +1,530 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "oit_linked_lists.h"
+#include <algorithm>
+
+OITLinkedLists::OITLinkedLists()
+{
+}
+
+OITLinkedLists::~OITLinkedLists()
+{
+	if (!device)
+	{
+		return;
+	}
+
+	vkDestroyPipeline(get_device().get_handle(), combine_pipeline, nullptr);
+	vkDestroyPipeline(get_device().get_handle(), background_pipeline, nullptr);
+	vkDestroyPipeline(get_device().get_handle(), gather_pipeline, nullptr);
+	vkDestroyPipelineLayout(get_device().get_handle(), pipeline_layout, nullptr);
+
+	vkDestroyDescriptorPool(get_device().get_handle(), descriptor_pool, VK_NULL_HANDLE);
+	vkDestroyDescriptorSetLayout(get_device().get_handle(), descriptor_set_layout, nullptr);
+
+	destroy_sized_objects();
+
+	instance_data.reset();
+	scene_constants.reset();
+
+	vkDestroySampler(get_device().get_handle(), background_texture.sampler, nullptr);
+	object.reset();
+}
+
+bool OITLinkedLists::prepare(const vkb::ApplicationOptions &options)
+{
+	if (!ApiVulkanSample::prepare(options))
+	{
+		return false;
+	}
+
+	camera.type = vkb::CameraType::LookAt;
+	camera.set_position({0.0f, 0.0f, -4.0f});
+	camera.set_rotation({0.0f, 0.0f, 0.0f});
+	camera.set_perspective(60.0f, static_cast<float>(width) / static_cast<float>(height), 256.0f, 0.1f);
+
+	load_assets();
+	create_constant_buffers();
+	create_descriptors();
+	create_sized_objects(width, height);
+	create_pipelines();
+
+	update_scene_constants();
+	fill_instance_data();
+	update_descriptors();
+	build_command_buffers();
+
+	prepared = true;
+	return true;
+}
+
+bool OITLinkedLists::resize(const uint32_t width, const uint32_t height)
+{
+	destroy_sized_objects();
+	create_sized_objects(width, height);
+	update_descriptors();
+	ApiVulkanSample::resize(width, height);
+	return true;
+}
+
+void OITLinkedLists::render(float delta_time)
+{
+	if (!prepared)
+	{
+		return;
+	}
+
+	ApiVulkanSample::prepare_frame();
+	submit_info.commandBufferCount = 1;
+	submit_info.pCommandBuffers    = &draw_cmd_buffers[current_buffer];
+	VK_CHECK(vkQueueSubmit(queue, 1, &submit_info, VK_NULL_HANDLE));
+	ApiVulkanSample::submit_frame();
+
+	if (camera_auto_rotation)
+	{
+		camera.rotate({delta_time * 5.0f, delta_time * 5.0f, 0.0f});
+	}
+	update_scene_constants();
+}
+
+void OITLinkedLists::request_gpu_features(vkb::PhysicalDevice &gpu)
+{
+	if (gpu.get_features().fragmentStoresAndAtomics)
+	{
+		gpu.get_mutable_requested_features().fragmentStoresAndAtomics = VK_TRUE;
+	}
+	else
+	{
+		throw std::runtime_error("This sample requires support for buffers and images stores and atomic operations in the fragment shader stage");
+	}
+
+	if (gpu.get_features().samplerAnisotropy)
+	{
+		gpu.get_mutable_requested_features().samplerAnisotropy = VK_TRUE;
+	}
+	else
+	{
+		throw std::runtime_error("This sample requires support for anisotropic sampling");
```

Comment:
Why does this sample require support for anisotropic sampling?
Could this just be optional?

Reply:
Ah! I had completely forgotten about this.
I enabled aniso because it was necessary to remove a validation error when using `ApiVulkanSample::load_texture`.
Other samples are enabling aniso as well (at least the 2-3 ones I have checked).

I think I fixed the issue in `ApiVulkanSample::load_texture`, but I am not too familiar with `ApiVulkanSample` code, so please have a look.
I removed aniso from the sample and it does not trigger any validation error anymore after the fix.